pkgs/development/compilers/hip/default.nix

```diff
@@ -154,6 +154,9 @@ stdenv.mkDerivation rec {
     "-DHIP_COMMON_DIR=${hip}"
     "-DROCCLR_PATH=${rocclr}"
     "-DHIP_VERSION_BUILD_ID=0"
+    "-DCMAKE_INSTALL_BINDIR=bin"
```

Comment:
This should be ideally fixed upstream, see https://github.com/jtojnar/cmake-snips#assuming-cmake_install_dir-is-relative-path

Reply:
Upstream hip or upstream cmake?
This doesn't seem to happen with all roc* projects (rocprim, hipcub), but I have observed 3 others create ${nix_path}/${package_name}/${nix_path}.
For hip, this seems to result in bad symlinks instead.
The problem seems to be nix-specific in that we use /nix/store to store package roots, and amd's cmake projects in particular end up taking the full path instead of a relative path.

Reply:
Hip, they are making incorrect assumptions:

- https://github.com/ROCm-Developer-Tools/hipamd/blob/928684dfed0fb4a8a14c6d5d79014324f3218d82/CMakeLists.txt#L44
- https://github.com/ROCm-Developer-Tools/hipamd/blob/928684dfed0fb4a8a14c6d5d79014324f3218d82/CMakeLists.txt#L280-L282
- https://github.com/ROCm-Developer-Tools/hipamd/blob/928684dfed0fb4a8a14c6d5d79014324f3218d82/CMakeLists.txt#L363-L364
- https://github.com/ROCm-Developer-Tools/hipamd/blob/928684dfed0fb4a8a14c6d5d79014324f3218d82/CMakeLists.txt#L475

Reply:
Here's an issue for now. https://github.com/ROCm-Developer-Tools/hipamd/issues/55